sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClientBuilder.java

```diff
@@ -318,8 +318,20 @@ public EncryptedBlobClientBuilder credential(TokenCredential credential) {
      * @throws NullPointerException If {@code sasToken} is {@code null}.
      */
     public EncryptedBlobClientBuilder sasToken(String sasToken) {
-        this.sasTokenCredential = new SasTokenCredential(Objects.requireNonNull(sasToken,
-            "'sasToken' cannot be null."));
+        return this.sasToken(new AzureSasCredential(Objects.requireNonNull(sasToken,
+            "'sasToken' cannot be null.")));
+    }
+
+    /**
+     * Sets the SAS token used to authorize requests sent to the service.
+     *
+     * @param sasToken The SAS token to use for authenticating requests.
+     * @return the updated EncryptedBlobClientBuilder
+     * @throws NullPointerException If {@code sasToken} is {@code null}.
+     */
+    public EncryptedBlobClientBuilder sasToken(AzureSasCredential sasToken) {
```

Comment:
Should this be called `credential` to go along with the other APIs accepting `*Credential` types?

Reply:
Do you think method name should be `sasToken` or `azureSasCredential`? I made it `sasToken` as its closest relative is already named like that and I assume we'd like them to be overloads.

Reply:
adjusted.